ksqldb-engine/src/main/java/io/confluent/ksql/engine/TransientQueryCleanupService.java

```diff
@@ -100,9 +110,18 @@ public void setQueryRegistry(final QueryRegistry queryRegistry) {
     this.queryRegistry = queryRegistry;
   }
 
-  @SuppressFBWarnings(value = "EI_EXPOSE_REP2")
   public void setLocalCommandsQueryAppIds(final Set<String> ids) {
-    this.localCommandsQueryAppIds = Optional.of(ids);
+    localCommandsQueryAppIds = ImmutableSet.copyOf(ids);
+    final ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+    for (String id : ids) {
+      final Matcher matcher = leakedAppIdPattern.matcher(id);
+      if (matcher.find()) {
+        // Get the query id and prepend it with prefix to get the incorrect topic pattern to match
```

Comment:
I'm thinking aloud if such comments shouldn't really be part of DEBUG or TRACE log level?

Reply:
I can do this